frontend/src/metabase-lib/lib/DimensionOptions/DimensionOptions.ts

```diff
@@ -36,7 +36,9 @@ export default class DimensionOptions {
   }
 
   sections({ extraItems = [] } = {}): DimensionOptionsSection[] {
-    const [dimension] = this.dimensions;
+    const dimension =
+      this.dimensions.find(dimension => !dimension.isExpression()) ??
+      this.dimensions[0];
```

Comment:
I'm not even sure if it's possible to have a list of dimensions with only expression dimensions, but just in case, we'll fall back to the first dimension.